### PR TITLE
br: multi download and parse metafile

### LINKED ISSUE #50772: br restore is very slow when there are many metafiles
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. snapshot restore

### 2. What did you expect to see? (Required)
1. success
### 3. What did you see instead (Required)
1. download and parse the metafiles slowly
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #50772

Problem Summary:
br restore is very slow when there are many metafiles
### What changed and how does it work?
multi download and parse metafile
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

1 hours -> 20 minutes

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/metautil/metafile.go [hunk 1]
```diff
@@ -116,32 +120,42 @@ func walkLeafMetaFile(
...
+			content, err := storage.ReadFile(ectx, node.Name)
+			if err != nil {
+				return errors.Trace(err)
+			}
 
-		decryptContent, err := Decrypt(content, cipher, node.CipherIv)
-		if err != nil {
-			return errors.Trace(err)
-		}
+			decryptContent, err := Decrypt(content, cipher, node.CipherIv)
```
Comment (3pointer) [inline br/pkg/metautil/metafile.go:133]: do we need a copy of cipher?

**Review by 3pointer — COMMENTED** [review]
LGTM

File: br/pkg/metautil/metafile.go (hunk 1 above)
Comment (Leavrth) [reply br/pkg/metautil/metafile.go:133]: it is a readonly data.

**Review by 3pointer — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**